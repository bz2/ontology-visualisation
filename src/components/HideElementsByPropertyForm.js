/* eslint react/jsx-key:0 */
import React from 'react'
import { connect } from 'redux-zero/react'
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next'
import { SelectButton } from 'primereact/selectbutton'
import { Button } from 'primereact/button'
import { InputText } from 'primereact/inputtext'
import { Dropdown } from 'primereact/dropdown'
import { orderBy } from 'lodash'
import actions from '../store/actions'
import {
  DEFAULT_HIDDEN_ELEMENT_PROPERTY,
  DEFAULT_HIDDEN_ELEMENT_SUBPROPERTY,
  EDGE_PROPERTIES_DROPDOWN
} from '../constants/graph'

const HideElementsByPropertyForm = ({
  elementType,
  index,
  elementProperties,
  annotationProperties,
  setProperty,
  annotationPropertiesDatasets,
}) => {
  const { t } = useTranslation()

  const elementProperty = elementProperties[index]

  const {
    type,
    properties
  } = elementProperty

  const logicButtons = [{
    value: 'and',
    label: t('and'),
    icon: 'pi-plus'
  }, {
    value: 'or',
    label: t('or'),
    icon: 'pi-minus'
  }]

  const operationButtons = [{
    value: 'contains',
    label: t('contains')
  }, {
    value: 'equal',
    label: t('equal')
  }, {
    value: 'notContains',
    label: t('notContains')
  }, {
    value: 'notEqual',
    label: t('notEqual')
  }]

  const itemTemplate = (option) => (
    <span className="graph-options-selection-row-select-option">
      <i className={`pi ${option.icon}`} />
      {` ${option.label}`}
    </span>
  )

  const annotationPropertyList = elementType === 'node' ? orderBy([
    ...annotationProperties,
    ...annotationPropertiesDatasets
  ], ['label'], ['asc']) : EDGE_PROPERTIES_DROPDOWN

  return (
    <div className="graph-options-property-form">
      <div
        className="graph-options-select-button"
      >
        <label htmlFor={`logic-select-${index}`}>
          {t('chooseLogic')}
        </label>
        <SelectButton
          id={`logic-select-${index}`}
          className="logic-select"
          value={type}
          options={logicButtons}
          onChange={(e) => setProperty({
            ...elementProperties,
            [index]: {
              ...elementProperties[index],
              type: e.value
            }
          })}
          itemTemplate={itemTemplate}
        />
      </div>

      {
        Object.keys(properties).length > 0
        && Object.keys(properties).map((propertyIndex) => {
          const {
            property,
            operation,
            value
          } = properties[propertyIndex]

          return (
            <div
              className="graph-options-property-form-property"
            >
              <div className="graph-options-property-form-row m-t-10">
                <label htmlFor={`filter-${type}-${index}-${propertyIndex}`}>
                  {t('chooseFilter')}
                </label>

                <Dropdown
                  id={`filter-${type}-${index}-${propertyIndex}`}
                  value={property}
                  options={annotationPropertyList}
                  filter
                  onChange={(e) => setProperty({
                    ...elementProperties,
                    [index]: {
                      ...elementProperties[index],
                      properties: {
                        ...elementProperties[index].properties,
                        [propertyIndex]: {
                          ...elementProperties[index].properties[propertyIndex],
                          property: e.value
                        }
                      }
                    }
                  })}
                  className="property-select"
                  placeholder={t('selectProperty')}
                />
              </div>

              <div className="graph-options-property-form-row m-t-10">
                <Dropdown
                  value={operation}
                  className="operation-select"
                  options={operationButtons}
                  filter
                  onChange={(e) => setProperty({
                    ...elementProperties,
                    [index]: {
                      ...elementProperties[index],
                      properties: {
                        ...elementProperties[index].properties,
                        [propertyIndex]: {
                          ...elementProperties[index].properties[propertyIndex],
                          operation: e.value
                        }
                      }
                    }
                  })}
                  placeholder={t('selectLogic')}
                />
              </div>

              <div className="graph-options-property-form-row m-t-10">
                <InputText
                  className="property-text-input value-input"
                  value={value}
                  placeholder={t('insertText')}
                  onChange={(e) => setProperty({
                    ...elementProperties,
                    [index]: {
                      ...elementProperties[index],
                      properties: {
                        ...elementProperties[index].properties,
                        [propertyIndex]: {
                          ...elementProperties[index].properties[propertyIndex],
                          value: e.target.value
                        }
                      }
                    }
                  })}
                />
              </div>

              <div className="graph-options-property-form-row m-t-10">
                <div className="network-styling-property-form-row">
                  <span className="p-buttonset add-delete-buttons">
                    <Button
                      label={t('delete')}
                      tooltip={t('delete')}
                      tooltipOptions={{
                        position: 'top'
                      }}
                      className="p-button-warning delete-property"
                      onClick={() => {
                        const newProperties = JSON.parse(JSON.stringify(elementProperties))
                        delete newProperties[index].properties[propertyIndex]

                        if (Object.keys(newProperties[index].properties).length === 0) {
                          newProperties[index].properties = { 0: DEFAULT_HIDDEN_ELEMENT_SUBPROPERTY }
                        }

                        setProperty(newProperties)
                      }}
                    />

                    <Button
                      label={t('add')}
                      className="add-property"
                      onClick={() => {
                        const newProperties = JSON.parse(JSON.stringify(elementProperties))
                        const nextIndex = Object.keys(newProperties[index].properties).length

                        newProperties[index].properties[nextIndex] = DEFAULT_HIDDEN_ELEMENT_SUBPROPERTY

                        setProperty(newProperties)
                      }}
                    />
                  </span>
                </div>
              </div>
            </div>
          )
        })
      }

      <div className="graph-options-button-wrapper">
        <Button
          label={t('delete')}
          tooltip={t('delete')}
          tooltipOptions={{
            position: 'top'
          }}
          iconPos="right"
          className="graph-options-button graph-options-button-delete"
          icon="pi pi-trash"
          onClick={() => {
            const newProperties = JSON.parse(JSON.stringify(elementProperties))
            delete newProperties[index]

            setProperty(
              Object.keys(newProperties).length > 0
                ? newProperties
                : {
                  0: DEFAULT_HIDDEN_ELEMENT_PROPERTY
                }
            )
          }}
        />
      </div>

    </div>
  )
}

HideElementsByPropertyForm.propTypes = {
  setProperty: PropTypes.func.isRequired,
  annotationProperties: PropTypes.arrayOf(PropTypes.shape).isRequired,
  annotationPropertiesDatasets: PropTypes.arrayOf(PropTypes.shape).isRequired,
  elementType: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  elementProperties: PropTypes.arrayOf(PropTypes.shape).isRequired,
}

const mapToProps = ({
  annotationProperties,
  annotationPropertiesDatasets
}) => ({
  annotationProperties,
  annotationPropertiesDatasets
})

export default connect(
  mapToProps,
  actions
)(HideElementsByPropertyForm)
