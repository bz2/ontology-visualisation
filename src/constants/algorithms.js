export const ALGO_TYPE_FULL = 'full'
export const ALGO_TYPE_NEIGHBOURHOOD = 'neighbourhood'
export const ALGO_TYPE_SHORTEST_PATH = 'shortest-path'
export const ALGO_TYPE_BOUNDING_BOX = 'bounding-box'
export const ALGO_TYPE_NODES_FILTER = 'nodes-filter'
export const ALGO_TYPE_EDGES_FILTER = 'edges-filter'
