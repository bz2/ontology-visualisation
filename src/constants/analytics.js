export const USERBACK_ACCESS_TOKEN = '29230|42198|CVJlhtfkaYN3J3xP2x200ghJ1'
export const USERBACK_COOKIE = 'ubwc'
export const USERBACK_WIDGET = 'https://static.userback.io/widget/v1.js'
export const GA_ID = 'UA-195025859-1'
