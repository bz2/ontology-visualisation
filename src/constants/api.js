export const API_ENDPOINT = `${process.env.endpointUrl}/api` // 'http://localhost:8080/api'

export const GET_ONTOLOGY = `${API_ENDPOINT}/ontology`
export const GET_GRAPH = `${API_ENDPOINT}/graph`
export const GET_GRAPH_QUERY = `${GET_GRAPH}/query`

export const GET_EDGES = `${GET_GRAPH}/edges`
export const GET_EDGE = `${GET_GRAPH}/edges/{id}`

export const GET_NODES = `${GET_GRAPH}/nodes`
export const GET_NODE = `${GET_GRAPH}/nodes/{id}`

// export const GET_PROPERTIES = `${GET_GRAPH}/properties`
// export const GET_PROPERTIES_OBJECTS = `${GET_PROPERTIES}/objects`
// export const GET_PROPERTIES_OBJECT = `${GET_PROPERTIES}/objects/{id}`
// export const GET_PROPERTIES_ANNOTATIONS = `${GET_PROPERTIES}/annotations`
// export const GET_PROPERTIES_ANNOTATION = `${GET_PROPERTIES}/annotations/{id}`
export const GET_PROPERTIES = `${GET_ONTOLOGY}/properties`
export const GET_PROPERTIES_OBJECTS = `${GET_PROPERTIES}/objects`
export const GET_PROPERTIES_OBJECT = `${GET_PROPERTIES}/objects/{id}`
export const GET_PROPERTIES_ANNOTATIONS = `${GET_PROPERTIES}/annotations`
export const GET_PROPERTIES_ANNOTATION = `${GET_PROPERTIES}/annotations/{id}`

export const AUTH_SIGN_IN = `${API_ENDPOINT}/login`
