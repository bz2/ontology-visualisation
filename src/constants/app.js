export const APP_NAME = 'OntoPop'
