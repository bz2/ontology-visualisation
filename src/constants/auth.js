export const AUTH_COOKIE = 'ontology_viz_auth'
