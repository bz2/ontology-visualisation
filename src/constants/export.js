export const EXPORT_JPG = 'jpeg'
export const EXPORT_PNG = 'png'
export const EXPORT_PDF = 'pdf'
export const EXPORT_CSV = 'csv'
export const EXPORT_OWL = 'owl'

export const EXPORT_GRAPH_OPTIONS = [
  EXPORT_JPG,
  EXPORT_PNG,
  EXPORT_PDF
]

export const EXPORT_DATA_OPTIONS = [
  EXPORT_CSV,
  EXPORT_OWL
]
