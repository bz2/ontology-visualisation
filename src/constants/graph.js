export const NODE_BACKGROUND = '#adefd1' // 'adefd1' // '03a9f4' // '2B7CE9',
export const NODE_BORDER = '#011e41' // 'D2E5FF',
export const NODE_BORDER_WIDTH = 1
export const CLICK_NODE_BACKGROUND = '#ffed00' // '#abd6df'
export const HIGHLIGHT_NODE_BACKGROUND = '#ffef00'
export const HIGHLIGHT_NODE_BORDER = '#009688'
export const HOVER_NODE_BACKGROUND = '#f2f2f2'
export const HOVER_NODE_BORDER = '#607d8b'
export const SELECTED_NODE_COLOR = '#ff6f61'
export const SPIDERABLE_NODE_BORDER_COLOR = '#ff6f61'
export const SPIDERABLE_NODE_BORDER_WIDTH = 2
export const NODE_TEXT_COLOR = '#000000'
export const NODE_DEFAULT_SHAPE = 'circle'
export const EDGE_COLOR = '#070b11'
export const EDGE_COLOR_SELECTED = '#03a9f4'
export const EDGE_COLOR_HIGHLIGHTED = '#9c27b0'

export const FONT_ALIGNMENT_OPTIONS = [
  { icon: 'pi pi-align-left', value: 'left' },
  { icon: 'pi pi-align-right', value: 'right' },
  { icon: 'pi pi-align-center', value: 'center' },
  { icon: 'pi pi-align-justify', value: 'justify' }
]

export const NODE_SHAPES_AFFECTED_BY_SIZE = [
  'diamond', 'dot', 'star', 'triangle', 'triangleDown', 'hexagon', 'square'
]

export const NODE_SHAPES = [
  ...NODE_SHAPES_AFFECTED_BY_SIZE,
  'ellipse',
  'circle',
  'database',
  'box',
  'text'
]

export const FILTER_TYPE_OPTIONS = ['contains', 'equal']

export const NODE_EDGE_BY_PROPERTY_STYLING_DEFAULT_OBJECT = {
  property: undefined,
  filterType: 'equal',
  filterValue: '',
  styleType: '',
  styleValue: ''
}

export const SUBCLASS_EDGE_STYLING_DEFAULT_OBJECT = {
  filterType: 'equal',
  filterValue: 'subClassOf',
  styleType: 'stylingEdgeLineStyle',
  styleValue: true,
  property: 'rdfsLabel'
}

export const FONT_ALIGNMENT_TEMPLATE = (option) => <i className={option.icon} />

export const NODE_STYLING_PROPERTIES = {
  stylingNodeSize: {
    type: 'number',
    defaultValue: 25,
    min: 1,
    max: 1000,
    step: 1,
    label: 'stylingNodeSize'
  },
  stylingNodeBorder: {
    type: 'number',
    defaultValue: 1,
    min: 1,
    max: 10,
    step: 0.5,
    label: 'stylingNodeBorder'
  },
  stylingNodeTextColor: {
    type: 'color',
    defaultValue: NODE_TEXT_COLOR.replace('#', ''),
    label: 'stylingNodeTextColor'
  },
  stylingNodeBorderSelected: {
    type: 'number',
    defaultValue: 2,
    min: 1,
    max: 10,
    step: 0.5,
    label: 'stylingNodeBorderSelected'
  },
  stylingNodeBorderColor: {
    type: 'color',
    defaultValue: NODE_BORDER.replace('#', ''),
    label: 'stylingNodeBorderColor'
  },
  stylingNodeBackgroundColor: {
    type: 'color',
    defaultValue: NODE_BACKGROUND.replace('#', ''),
    label: 'stylingNodeBackgroundColor'
  },
  stylingNodeHighlightBorderColor: {
    type: 'color',
    defaultValue: HIGHLIGHT_NODE_BORDER.replace('#', ''),
    label: 'stylingNodeHighlightBorderColor'
  },
  stylingNodeHighlightBackgroundColor: {
    type: 'color',
    defaultValue: CLICK_NODE_BACKGROUND.replace('#', ''),
    label: 'stylingNodeHighlightBackgroundColor'

  },
  stylingNodeHoverBackgroundColor: {
    type: 'color',
    defaultValue: HOVER_NODE_BACKGROUND.replace('#', ''),
    label: 'stylingNodeHoverBackgroundColor'

  },
  stylingNodeHoverBorderColor: {
    type: 'color',
    defaultValue: HOVER_NODE_BORDER.replace('#', ''),
    label: 'stylingNodeHoverBorderColor'
  },
  stylingNodeShape: {
    type: 'dropdown',
    defaultValue: NODE_DEFAULT_SHAPE,
    options: NODE_SHAPES,
    needsI18n: true,
    label: 'stylingNodeShape',
    placeholder: 'nodeShapeInstructions'
  },
  stylingNodeTextFontSize: {
    type: 'number',
    defaultValue: 12,
    min: 1,
    max: 200,
    step: 1,
    label: 'stylingNodeTextFontSize'
  },
  stylingNodeTextFontAlign: {
    type: 'selectButton',
    defaultValue: 'center',
    options: FONT_ALIGNMENT_OPTIONS,
    label: 'stylingNodeTextFontAlign',
    itemTemplate: FONT_ALIGNMENT_TEMPLATE
  },
}

export const EDGE_LINE_STYLE_TEMPLATE = (option) => <i className={option.icon} />

export const EDGE_LINE_STYLE_OPTIONS = [
  { icon: 'pi pi-ellipsis-h', value: true },
  { icon: 'pi pi-minus', value: false }
]

export const EDGE_ALIGNMENT_OPTIONS = ['horizontal', 'top', 'middle', 'bottom']

export const EDGE_STYLING_PROPERTIES = {
  stylingEdgeLength: {
    type: 'number',
    defaultValue: 250,
    min: 1,
    max: 1000,
    step: 1,
    label: 'stylingEdgeLength'
  },
  stylingEdgeWidth: {
    type: 'number',
    defaultValue: 1,
    min: 1,
    max: 20,
    step: 1,
    label: 'stylingEdgeWidth'
  },
  stylingEdgeLineColor: {
    type: 'color',
    defaultValue: EDGE_COLOR.replace('#', ''),
    label: 'stylingEdgeLineColor'
  },
  stylingEdgeLineColorHighlight: {
    type: 'color',
    defaultValue: EDGE_COLOR_HIGHLIGHTED.replace('#', ''),
    label: 'stylingEdgeLineColorHighlight'
  },
  stylingEdgeLineColorHover: {
    type: 'color',
    defaultValue: EDGE_COLOR.replace('#', ''),
    label: 'stylingEdgeLineColorHover'
  },
  stylingEdgeLineStyle: {
    type: 'selectButton',
    defaultValue: false,
    options: EDGE_LINE_STYLE_OPTIONS,
    label: 'stylingEdgeLineStyle',
    itemTemplate: EDGE_LINE_STYLE_TEMPLATE
  },
  stylingEdgeTextSize: {
    type: 'number',
    defaultValue: 1,
    min: 1,
    max: 200,
    step: 1,
    label: 'stylingEdgeTextSize'
  },
  stylingEdgeTextColor: {
    type: 'color',
    defaultValue: CLICK_NODE_BACKGROUND.replace('#', ''),
    label: 'stylingEdgeTextColor'
  },
  stylingEdgeTextAlign: {
    type: 'selectButton',
    defaultValue: 'horizontal',
    options: EDGE_ALIGNMENT_OPTIONS,
    label: 'stylingEdgeTextAlign',
    needsI18n: true,
  },
}

export const SUB_CLASS_OF_ID = 'http://www.w3.org/2000/01/rdf-schema#subclassof'
export const SUB_CLASS_OF_LABEL = 'subClassOf'

export const SUB_CLASS_OF_OBJECT = {
  id: SUB_CLASS_OF_ID,
  label: SUB_CLASS_OF_LABEL,
  value: SUB_CLASS_OF_ID
}

export const LOW_LEVEL_PROPERTIES = [
  'rdfAbout',
  'rdfsLabel',
  'skosDefinition',
  'skosComment'
]

export const MINIMAL_LOW_LEVEL_PROPERTIES = [
  'rdfAbout',
  'rdfsLabel',
]

export const REQUIRED_PROPERTIES = [
  'rdfAbout',
]

export const UNIQUE_PROPERTY = 'rdfAbout'
export const LABEL_PROPERTY = 'rdfsLabel'
export const SUBCLASSOF_PROPERTY = 'rdfsSubClassOf'
export const OWL_ANNOTATION_PROPERTIES = 'owlAnnotationProperties'
export const OWL_RESTRICTION = 'owlRestriction'

export const REQUIRED_PREDICATES = [
  'rdfsSubClassOf',
  'http://www.w3.org/2000/01/rdf-schema#subclassof',
  'http://www.w3.org/2002/07/owl#topObjectProperty'
]

export const GRAPH_VERSION_STRUCTURE = {
  classesFromApi: {},
  objectPropertiesFromApi: {},
  classesFromApiBackup: {},
  objectPropertiesFromApiBackup: {},
  deletedNodes: [],
  addedNodes: [],
  updatedNodes: [],
  deletedEdges: [],
  addedEdges: [],
  updatedEdges: [],
  deletedConnections: [],
  addedConnections: [],
}

export const PROPERTIES_TO_IGNORE = [
  'id',
  'label',
  'x',
  'y'
]
