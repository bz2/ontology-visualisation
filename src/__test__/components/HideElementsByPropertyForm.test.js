import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import HideElementsByPropertyForm from '../../components/HideElementsByPropertyForm'
import { DEFAULT_HIDDEN_ELEMENT_PROPERTY } from '../../constants/graph'

const setup = ({
  elementType
}) => {
  const props = {
    elementType,
    index: 0,
    elementProperties: { 0: DEFAULT_HIDDEN_ELEMENT_PROPERTY },
    annotationProperties: [{
      label: 'rdfsLabel',
      value: 'rdfsLabel',
    }],
    setProperty: jest.fn(),
    annotationPropertiesDatasets: [{
      label: 'upperOntology',
      value: 'upperOntology',
    }],
  }

  const component = shallow(<HideElementsByPropertyForm {...props} />)

  return {
    component,
    props
  }
}

describe('HideElementsByPropertyForm', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should match snapshot when node', () => {
    const {
      component
    } = setup({
      elementType: 'node'
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when edge', () => {
    const {
      component
    } = setup({
      elementType: 'edge'
    })

    expect(toJson(component)).toMatchSnapshot()
  })
})
