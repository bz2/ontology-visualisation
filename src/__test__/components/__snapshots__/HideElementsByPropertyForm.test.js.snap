// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HideElementsByPropertyForm should match snapshot when edge 1`] = `
<Connect
  actions={
    Object {
      "addNumber": [Function],
      "addSubValueToObject": [Function],
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "setStoreState": [Function],
      "toggleFromArrayInKey": [Function],
      "toggleFromSubArray": [Function],
    }
  }
  annotationProperties={
    Array [
      Object {
        "label": "rdfsLabel",
        "value": "rdfsLabel",
      },
    ]
  }
  annotationPropertiesDatasets={
    Array [
      Object {
        "label": "upperOntology",
        "value": "upperOntology",
      },
    ]
  }
  elementProperties={
    Object {
      "0": Object {
        "properties": Object {
          "0": Object {
            "operation": "includes",
            "property": "",
            "value": "",
          },
        },
        "type": "and",
      },
    }
  }
  elementType="edge"
  index={0}
  mapToProps={[Function]}
  setProperty={[MockFunction]}
>
  <Component />
</Connect>
`;

exports[`HideElementsByPropertyForm should match snapshot when node 1`] = `
<Connect
  actions={
    Object {
      "addNumber": [Function],
      "addSubValueToObject": [Function],
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "setStoreState": [Function],
      "toggleFromArrayInKey": [Function],
      "toggleFromSubArray": [Function],
    }
  }
  annotationProperties={
    Array [
      Object {
        "label": "rdfsLabel",
        "value": "rdfsLabel",
      },
    ]
  }
  annotationPropertiesDatasets={
    Array [
      Object {
        "label": "upperOntology",
        "value": "upperOntology",
      },
    ]
  }
  elementProperties={
    Object {
      "0": Object {
        "properties": Object {
          "0": Object {
            "operation": "includes",
            "property": "",
            "value": "",
          },
        },
        "type": "and",
      },
    }
  }
  elementType="node"
  index={0}
  mapToProps={[Function]}
  setProperty={[MockFunction]}
>
  <Component />
</Connect>
`;
